### ISSUE TITLE
`useCollection().add` incorrectly requires ID field

### ISSUE THREAD

**Issue by daneden** [nandorojo/swr-firestore#15]
The `add` function returned from `useCollection` uses the same type signature as regular ol' `Document`, which requires an ID:

https://github.com/nandorojo/swr-firestore/blob/0ceda73d8fe1661a52120cdf3097a59e1b6952f4/src/hooks/use-swr-collection.ts#L8

For `add`ing to a collection, IDs are generated automatically by Firestore. The type signature should exclude that field to prevent write errors/warnings.

**Comment by nandorojo** [comment]
Thanks so much for the issue! To make sure I understand — do you mean that when you call `add`, the object you’re passing to it is requesting an `id` field? If that’s the case, it’s definitely a bug, and I’ll fix that!

https://github.com/nandorojo/swr-firestore/blob/0ceda73d8fe1661a52120cdf3097a59e1b6952f4/src/hooks/use-swr-collection.ts#L325

This is the relevant line. I’ll take a look!

**Comment by nandorojo** [comment]
You’re right, I see it. This line is the problem:

https://github.com/nandorojo/swr-firestore/blob/0ceda73d8fe1661a52120cdf3097a59e1b6952f4/src/hooks/use-swr-collection.ts#L326

It should be `(data: Data | Data[])` instead of `Doc | Doc[]`.

I’m on my phone at the moment, but can fix + publish that shortly when I’m at my laptop.

**Comment by daneden** [comment]
@nandorojo thanks for the fast response! I got around it for now by just using the `uid` library to generate an ID. Looking forward to the fix!

Also, thanks so much for this library—it’s removed a lot of boilerplate/finnicky API code on my new project!

**Comment by nandorojo** [comment]
@daneden You got it — thanks for the heads up on the bug. It’s always helpful when someone finds something that got by me.

And I love to hear that! Would love to see your project (and if it’s OS, it would be awesome to see how this lib is used in your code.)

**Comment by nandorojo** [comment]
This should be fixed in the new update `0.7.9-alpha.1`.

```sh
yarn add @nandorojo/swr-firestore@next

# or
npm i @nandorojo/swr-firestore@next
```

Could you try it out and let me know if that fixes it?

**Comment by nandorojo** [comment]
By the way – when I first started learning JS [in 2017](https://github.com/nandorojo/animateMePlz), I made a small jQuery library for for CSS animations. It was built on top of your animate.css library. Felt like a fun thing to share, now that you're using a library of mine.

https://github.com/nandorojo/animateMePlz

<img width="482" alt="Screen Shot 2020-06-17 at 5 09 32 PM" src="https://user-images.githubusercontent.com/13172299/84950715-52a3de80-b0bd-11ea-8c64-fedc1fd1acd8.png">

**Comment by daneden** [comment]
Nice—the fix works perfectly! Thanks for the fast response!

Closed by ca379da724c427e95acf188624cbb8a0f134e078

**Comment by nandorojo** [comment]
Sounds good, I’ll publish it to the main release version today.

**Comment by danoc** [comment]
Not sure if it was on purpose, but the change in `ca379da724c427e95acf188624cbb8a0f134e078` (released in `0.7.9-alpha.1`) also fixed an issue I was seeing where SWR was calling Firebase even if `path` is `null`. [Was fixed in this line](https://github.com/nandorojo/swr-firestore/blob/1d3edbb73f99c902ab3eb3abdb32174bc4c4de7e/src/hooks/use-swr-collection.ts#L235).

Thanks for the great library!

**Comment by nandorojo** [comment]
Yeah, I actually noticed that the `null` case was missing for `useCollection` while fixing this bug, so I fixed both in the same commit. 😅

I’ll get this published to the latest tag once I clean up some other stuff.

Glad you’re enjoying it, and thanks for letting me know about the `null` case!
